src/gt4py/stencil_builder.py

```diff
@@ -47,7 +47,7 @@ def __init__(
         self,
         definition_func: Union[StencilFunc, AnnotatedStencilFunc],
         *,
-        backend: Optional[Type["BackendType"]] = None,
+        backend: Optional[Union[str, Type["BackendType"]]] = None,
```

Comment:
Is that acceptable from a Python point of view? Or should we allow only backend name as was my initial idea? Then we need to update a lot of tests that pass the backend, not the name.